tests/unittests/executor/test_executor.py

```diff
@@ -0,0 +1,119 @@
+import sys
+
+import pytest
+
+from orion.executor.dask_backend import Dask
+from orion.executor.multiprocess_backend import PoolExecutor
+from orion.executor.single_backend import SingleExecutor
+
+
+def multiprocess(n):
+    return PoolExecutor(n, "multiprocess")
+
+
+def thread(n):
+    return PoolExecutor(n, "threading")
+
+
+backends = [thread, multiprocess, Dask, SingleExecutor]
+
+
+def function(a, b, c):
+    return a + b * c
+
+
+class BadException(Exception):
+    pass
+
+
+def bad_function(a, b, c):
+    raise BadException()
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_function(backend):
+    with backend(5) as executor:
+        future = executor.submit(function, 1, 2, c=3)
+        assert executor.wait([future]) == [7]
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_bad_function(backend):
+    with backend(5) as executor:
+        future = executor.submit(bad_function, 1, 2, 3)
+        with pytest.raises(BadException):
+            executor.wait([future])
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_async(backend):
+    with backend(5) as executor:
+        futures = [executor.submit(function, 1, 2, i) for i in range(10)]
+
+        total_task = len(futures)
+        results = executor.async_get(futures)
+
+        assert len(results) <= total_task, "Maybe not all tasks were completed"
```

Comment:
We should make sure at least some of the functions were executed, otherwise we are not testing the results properly and futures may not be removed.

Reply:
added assert to make sure we got some results and the finished future got popped